metronome/rocksdb/src/io/iohk/metronome/rocksdb/RocksDBStore.scala

```diff
@@ -0,0 +1,374 @@
+package io.iohk.metronome.rocksdb
+
+import cats._
+import cats.implicits._
+import cats.data.ReaderT
+import cats.effect.{Resource, Sync}
+import cats.free.Free.liftF
+import io.iohk.metronome.storage.{KVStore, KVStoreOp}
+import io.iohk.metronome.storage.KVStoreOp.{Put, Get, Delete}
+import java.util.concurrent.locks.ReentrantReadWriteLock
+import org.rocksdb.{
+  RocksDB,
+  WriteBatch,
+  WriteOptions,
+  ReadOptions,
+  Options,
+  DBOptions,
+  ColumnFamilyOptions,
+  ColumnFamilyDescriptor,
+  ColumnFamilyHandle,
+  BlockBasedTableConfig,
+  BloomFilter,
+  CompressionType,
+  ClockCache
+}
+import scodec.Codec
+import scodec.bits.BitVector
+import scala.collection.mutable
+import java.nio.file.Path
+import scala.annotation.nowarn
+
+class RocksDBStore[F[_]: Sync](
```

Comment:
Hm, it may seem too much; I just removed all comments and `class RocksDB` is only 130 lines long, and out of that only `writeBatch` and `read` interact with the database, so I'm not sure what is there to factor out (1 `db.write` and 1 `db.get`).

Reply:
I can move locks to a support trait, perhaps that would reduce the load and even be reusable if you want other types of compilers that don't interact with each other like this to live in different classes, but think splitting out a "bytes only" interface may even make things more difficult to understand, since the building of batching happens in the compilers, it's not super meaningful on its own.

Reply:
Split it up a bit.